sdk/core/core-rest-pipeline/src/restError.ts

```diff
@@ -72,8 +72,8 @@ export class RestError extends Error {
     this.name = "RestError";
     this.code = options.code;
     this.statusCode = options.statusCode;
-    this.request = options.request;
-    this.response = options.response;
+    this.request = options.request && errorSanitizer.sanitizePipelineRequest(options.request);
```

Comment:
Should we need to do anything about toJSON at all as well for the unit tests or is that covered in the custom string method?

Reply:
Shouldn't be necessary to do anything extra for toJSON. Since the actual properties on the object are being sanitized with this change, it follows that the toJSON representation of the object will also be clean.